salt/modules/cmdmod.py

```diff
@@ -680,6 +682,23 @@ def _run(
             ]
         except ValueError:
             raise SaltInvocationError("success_retcodes must be a list of integers")
+
+    if success_stdout is None:
+        success_stdout = []
+    else:
+        try:
+            success_stdout = [i for i in salt.utils.args.split_input(success_stdout)]
+        except ValueError:
+            raise SaltInvocationError("success_stdout must be a list of integers")
```

Comment:
Huh. Wait a minute - this exception specifies integers. Either the wrong exception is happening here, or the implementation is incorrect (e.g. that should've been `int(i) for i in ...` up there)

Reply:
I thought this was confusing messaging, but I was assuming @garethgreenaway had the right implementation. It was already reviewed, approved, and merged earlier.

Reply:
The idea was to have a successfully retcode returned if there was a match or partial match in either stdout and stderr when using cmdmod to run something.  The output could be multiple strings, hence building the list.

Reply:
So that error in the raise is wrong.  `success_stdout` should be a list of strings, not a list of ints.

Reply:
Ok, I see, this code was copied from `success_retcodes` above. But there, it needed to cast the items to `int`, cuz they are return codes, which is why there's a list comprehension on that one. In that case, I don't think either the list comprehension or the try/except is even necessary for `success_stdout` and `success_stderr`.

Reply:
I believe I addressed this, but please let me know if I missed the mark by removing the try/except.

Reply:
Looks good.